build-system/tasks/visual-diff.js

```diff
@@ -16,10 +16,74 @@
 'use strict';
 
 const argv = require('minimist')(process.argv.slice(2));
+const BBPromise = require('bluebird');
+const colors = require('ansi-colors');
+const fancyLog = require('fancy-log');
+const fs = require('fs');
 const gulp = require('gulp-help')(require('gulp'));
-const {execOrDie} = require('../exec');
+const JSON5 = require('json5');
+const path = require('path');
+const puppeteer = require('puppeteer');
+const request = BBPromise.promisify(require('request'));
+const sleep = require('sleep-promise');
+const tryConnect = require('try-net-connect');
+const {execOrDie, execScriptAsync} = require('../exec');
+const {FileSystemAssetLoader, Percy} = require('@percy/puppeteer');
 const {gitBranchName, gitCommitterEmail} = require('../git');
 
+// CSS widths: iPhone: 375, Pixel: 411, Desktop: 1400.
+const SNAPSHOT_OPTIONS = {widths: [375, 411, 1400]};
+const SNAPSHOT_EMPTY_BUILD_OPTIONS = {widths: [375]};
+const VIEWPORT_WIDTH = 1400;
+const VIEWPORT_HEIGHT = 100000;
+const HOST = 'localhost';
+const PORT = 8000;
+const BASE_URL = `http://${HOST}:${PORT}`;
+const WEBSERVER_TIMEOUT_RETRIES = 10;
+const NAVIGATE_TIMEOUT_MS = 12000;
+const CONFIGS = ['canary', 'prod'];
+const CSS_SELECTOR_TIMEOUT_MS = 5000;
+const AMP_RUNTIME_TARGET_FILES = [
+  'dist/amp.js', 'dist.3p/current/integration.js'];
+const BUILD_STATUS_URL = 'https://amphtml-percy-status-checker.appspot.com/status';
+const BUILD_PROCESSING_POLLING_INTERVAL_MS = 5 * 1000; // Poll every 5 seconds
+const BUILD_PROCESSING_TIMEOUT_MS = 15 * 1000; // Wait for up to 10 minutes
+const MASTER_BRANCHES_REGEXP = /^(?:master|release|canary|amp-release-.*)$/;
+const PERCY_BUILD_URL = 'https://percy.io/ampproject/amphtml/builds';
+
+/**
+ * Logs a message to the console.
+ *
+ * @param {string} mode
+ * @param {!Array<string>} messages
+ */
+function log(mode, ...messages) {
+  switch (mode) {
+    case 'verbose':
+      if (process.env.TRAVIS) {
+        return;
+      }
+      messages.unshift(colors.green('VERBOSE:'));
+      break;
+    case 'info':
+      messages.unshift(colors.green('INFO:'));
+      break;
+    case 'warning':
+      messages.unshift(colors.yellow('YELLOW:'));
```

Comment:
@danielrozenberg Another much delayed review comment: This should be `WARNING`, not `YELLOW` :)